src/conf_mode/load-balancing-inbound-ipvsadm.py

```diff
@@ -0,0 +1,150 @@
+            if virtual_ip not in vip_list:
+                vip_list.append(virtual_ip)
+        
+        for rule, rule_config in config['rule'].items():
+            if 'disable' in rule_config:
+                continue
+            
+            err_msg = f'Error in the configuration of rule {rule}. '
+            
+            if is_ipv4(rule_config['virtual_ip']):
+                ip_type = 4
+            elif is_ipv6(rule_config['virtual_ip']):
+                ip_type = 6
+                
+            backend_server_list = []
+            if 'backend' in rule_config:
+                for backend, backend_config in rule_config['backend'].items():
+                    if 'disable' in backend_config:
+                        return None
+                        
+                    if 'backend_ip' not in backend_config:
+                        raise ConfigError(err_msg + 'The backend server IP must exist!')
+                    
+                    backend_ip = backend_config['backend_ip']
+                    virtual_port = rule_config['virtual_port']
+                    backend_server = None
+                    
+                    backend_server = f'{bracketize_ipv6(backend_ip)}:{virtual_port}'
+                    
+                    if backend_server not in backend_server_list:
+                        backend_server_list.append(backend_server)
+                    else:
+                        raise ConfigError(err_msg + 'Do not allow duplicate backend servers!')
+                    
+                    if backend_ip in vip_list:
+                        raise ConfigError(err_msg + 'The back-end server cannot use VIP (virtual server IP)!')
+                        
+                    if (ip_type == 4 and not is_ipv4(backend_config['backend_ip'])) or \
+                       (ip_type == 6 and not is_ipv6(backend_config['backend_ip'])):
+                        raise ConfigError(err_msg + f'The IP of the backend server {backend} \
+                            of rule {rule} must be in the same address family as the virtual server IP!')
+            
+            if 'algorithm' not in rule_config and 'mode' in rule_config:
+                raise ConfigError(err_msg + 'mode and algorithm must be configured!')
+            
+            if dict_search('mode', rule_config) == 'tun' and dict_search('tun_option.type', rule_config) =='gue' and \
+               not dict_search('tun_option.port', rule_config):
+                raise ConfigError(err_msg + 'Port specification is required if GUE tunneling is used in lvs-tun mode!')
+    
+    return None
+
+def generate(config):
+    if not config:
+        return None
+    
+    render(lvs_config, 'load-balancing/ipvsadm.rules.tmpl', config, permission=0o644)
+
+def apply(config):
+    
+    if os.path.isfile(lvs_config):
+        if not config:
+            cmd(f'ipvsadm -C')
+        else:
+            cmd(f'ipvsadm -R < {lvs_config}')
+        os.unlink(lvs_config)
```

Comment:
Does always unlinking the configuration file make sense?

Reply:
done